plugin/kprom/kprom.go

```diff
@@ -360,33 +366,42 @@ func (m *Metrics) OnNewClient(client *kgo.Client) {
 		},
 		func() float64 { return float64(client.BufferedFetchRecords()) },
 	)
+
+	m.allMetricCollectors = append(m.allMetricCollectors,
```

Comment:
Is there a reason to not keep the old code, gated behind the new conditional below? This looks to add a slice that has the same amount of lines and requires the same updates whenever a new metric is added.

Reply:
I introduced this flag to keep the existing behaviour untouched.
But your suggestion of `nil` check on `Register` also will work.
I have updated the code in the latest commit.